flink-staging/flink-streaming/flink-storm-compatibility/src/main/java/org/apache/flink/stormcompatibility/api/FlinkTopology.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.flink.stormcompatibility.api;
+
+import org.apache.flink.api.common.JobExecutionResult;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+
+import backtype.storm.generated.StormTopology;
+
+
+
+
+
+/**
+ * {@link FlinkTopology} mimics a {@link StormTopology} and is implemented in terms of a
+ * {@link StreamExecutionEnvironment} . In contrast to a regular {@link StreamExecutionEnvironment}, a
+ * {@link FlinkTopology} cannot be executed directly, but must be handed over to a {@link FlinkLocalCluster},
+ * {@link FlinkSubmitter}, or {@link FlinkClient}.
+ * 
+ * @author mjsax
+ */
+class FlinkTopology extends StreamExecutionEnvironment {
+	/**
+	 * The corresponding {@link StormTopology} that is mimiced by this {@link FlinkTopology}.
+	 */
+	private final StormTopology stormTopology;
+	/**
+	 * The number of declared tasks for the whole program (ie, sum over all dops)
+	 */
+	private int numberOfTasks = 0;
+	
+	
+	
+	/**
+	 * Instantiates a new {@link FlinkTestTopology}.
+	 */
+	public FlinkTopology(final StormTopology stormTopology) {
+		super.setParallelism(1);
+		this.stormTopology = stormTopology;
+	}
+	
```

Comment:
Storm uses default parallelism of one, thus, we set it to one here. We do not want to break potential user assumptions about default dop.

It also works for different dop values; the user has to specify non-default dop when adding Spouts/Bolts to the topology.